streamad/model/OCSVM_Detector.py

```diff
@@ -0,0 +1,54 @@
+from sklearn.svm import OneClassSVM
+import pandas as pd
+from sklearn.preprocessing import StandardScaler
+import numpy as np
+from streamad.base.detector import BaseDetector
+
+
+class OCSVMDetector(BaseDetector):
+    def __init__(
+        self,
+        nu: float=0.5,
+        kernel="rbf",
+        window_length: int=10,
+    ):
+        """One-Class SVM Detector :cite:`conf/icde/KhelifatiKCHLH21`.
+
+        Args:
+            nu (float, optional): An upper bound on the fraction of training errors and a lower bound of the fraction of support vectors. Defaults to 0.5.
+            kernel (str, optional): The kernel type to be used in the algorithm. Defaults to "rbf".
+            window_length (int, optional): Length of sliding window. Defaults to 10.
+        """
+        super().__init__()
+        self.nu = nu
+        self.kernel = kernel
+        self.buf = []
+        self.count = 0
+        self.window_length = window_length

```

Comment:
It seems that you do not use the window length parameter to limit the buffer size? Plz correct me if I'm wrong.